nixos/modules/services/desktops/bonsaid.nix

```diff
@@ -0,0 +1,167 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  json = pkgs.formats.json { };
+  transitionType =
+    with lib;
+    types.submodule {
+      freeformType = json.type;
+      options.type = mkOption {
+        type = types.enum [
+          "delay"
+          "event"
+          "exec"
+        ];
+        description = ''
+          Type of transition. Determines how bonsaid interprets the other options in this transition.
+        '';
+      };
+      options.command = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = ''
+          Command to run when this transition is taken.
+          This is executed inline by `bonsaid` and blocks handling of any other events until completion.
+          To perform the command asynchronously, specify it like `[ "setsid" "-f" "my-command" ]`.
+
+          Only effects transitions with `type = "exec"`.
+        '';
+      };
+      options.delay_duration = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Nanoseconds to wait after the previous state change before performing this transition.
+          This can be placed at the same level as a `type = "event"` transition to achieve a
+          timeout mechanism.
+
+          Only effects transitions with `type = "delay"`.
+        '';
+      };
+      options.event_name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Name of the event which should trigger this transition when received by `bonsaid`.
+          Events are sent to `bonsaid` by running `bonsaictl -e <event_name>`.
+
+          Only effects transitions with `type = "event"`.
+        '';
+      };
+      options.transitions = mkOption {
+        type = types.listOf transitionType;
+        default = [ ];
+        description = ''
+          List of transitions out of this state.
+          If left empty, then this state is considered a terminal state and entering it will
+          trigger an immediate transition back to the root state (after processing side effects).
+        '';
+        visible = "shallow";
+      };
+    };
+  cfg = config.services.bonsaid;
+in
+{
+  meta.maintainers = with lib.maintainers; [ colinsane ];
+  meta.buildDocsInSandbox = false;
+
+  options.services.bonsaid = with lib; {
+    enable = mkEnableOption "bonsaid";
+    package = mkPackageOption pkgs "bonsai" { };
+    extraFlags = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = ''
+        Extra flags to pass to `bonsaid`, such as `[ "-v" ]` to enable verbose logging.
+      '';
+    };
+    settings = mkOption {
+      type = types.listOf transitionType;
+      default = [ ];
```

Comment:
Would just setting `services.bonsaid.enable = true;` be enough for the service to start and be useful with such default value? If not it's better to just drop it.
```suggestion
```

Reply:
the service should start with `settings` empty (untested), but it will surely be useless. good call to remove the default here.